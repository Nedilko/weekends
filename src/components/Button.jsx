import PropTypes from 'prop-types'

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
}

function Button({ onClick, title }) {
  return (
    <button
      className="w-16 mx-4 py-2 font-thin text-xl hover:scale-110 duration-200 ease-in-out text-slate-600 dark:text-gray-400"
      onClick={onClick}
    >
      {title}
    </button>
  )
}

export default Button
