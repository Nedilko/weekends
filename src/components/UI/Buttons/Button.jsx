import PropTypes from 'prop-types'

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
}

function Button({ onClick, title }) {
  return (
    <button
      className="mx-4 w-16 py-2 text-xl font-thin text-slate-600 duration-200 ease-in-out hover:scale-110 focus:scale-110 focus:outline-none dark:text-gray-400"
      onClick={onClick}
    >
      {title}
    </button>
  )
}

export default Button
