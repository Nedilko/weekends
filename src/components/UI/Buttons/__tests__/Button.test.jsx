import Button from '@UI/Buttons/Button'
import { render, screen, userEvent } from '@utils/test-utils'

describe('button is', () => {
  const clickHandler = vi.fn()
  it('in the document', () => {
    render(<Button title="sample" onClick={clickHandler} />)
    const button = screen.getByRole('button')
    expect(button).toBeInTheDocument()
    expect(button).toHaveTextContent('sample')
    expect(button).toMatchSnapshot()
  })
  it('clicked once', async () => {
    render(<Button title="sample" onClick={clickHandler} />)
    const user = userEvent.setup()
    const button = screen.getByRole('button')
    await user.click(button)
    expect(clickHandler).toHaveBeenCalledTimes(1)
  })
})
