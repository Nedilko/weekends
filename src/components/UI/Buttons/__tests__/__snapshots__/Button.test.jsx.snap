// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be in the document 1`] = `
<button
  class="mx-4 w-16 py-2 text-xl font-thin text-slate-600 duration-200 ease-in-out hover:scale-110 focus:scale-110 focus:outline-none dark:text-gray-400"
>
  sample
</button>
`;
